Docs: Add What's New in v9.1 (#53066)
grafana/grafana: docs/sources/whatsnew/whats-new-in-v9-1.md

---
_build:
  list: false
aliases:
  - /docs/grafana/latest/guides/whats-new-in-v9-1/
description: Feature and improvement highlights for Grafana v9.1
keywords:
  - grafana
  - new
  - documentation
  - '9.1'
  - release notes
title: What's new in Grafana v9.1
weight: -33
---

# What's new in Grafana v9.1

> We're excited to announce Grafana v9.1, with a variety of improvements that focus on Grafana's usability, performance, and security. Read on to learn about new options to share and embed dashboards, search and navigation enhancements, new panel options and additional authentication features. You can also find out more about new Single Sign-On and role-based access control options in Grafana Enterprise, and more. For details, see the complete [changelog](https://github.com/grafana/grafana/blob/master/CHANGELOG.md).

Reviewer: Are you using the line break best practice where one sentence is on each line? Is that something you want me to check for? 
```suggestion
We're excited to announce Grafana v9.1, with a variety of improvements that focus on Grafana's usability, performance, and security. 
Read on to learn about new options to share and embed dashboards, search and navigation enhancements, new panel options and additional authentication features. 
You can also find out more about new Single Sign-On and role-based access control options in Grafana Enterprise, and more. 
For details, see the complete [changelog](https://github.com/grafana/grafana/blob/master/CHANGELOG.md).
```

## OSS

### Auth and security

#### Grafana Service Accounts are generally available

Service accounts are an evolution in machine access within Grafana. You can create multiple API tokens per service account with independent expiration dates, and temporarily disable a service account without deleting it. These benefits make Service Accounts a more flexible way for Terraform and other apps to authenticate with Grafana.

> Service accounts also work with [role-based access control]({{< relref "../enterprise/access-control/" >}}) in [Grafana Enterprise]({{< relref "../enterprise/" >}}): you can improve security by granting service accounts specific roles to limit the functions they can perform. Service accounts have been in beta since Grafana v8.5, and during that time we've improved the UI and migration path from API keys, and made it possible to add service accounts to teams and inherit team permissions. To learn more about Service Accounts, see the [documentation]({{< relref "../administration/service-accounts" >}}).

Reviewer: Fixing a run-on sentence. 
```suggestion
Service accounts also work with [role-based access control]({{< relref "../enterprise/access-control/" >}}) in [Grafana Enterprise]({{< relref "../enterprise/" >}}). 
You can improve security by granting service accounts specific roles to limit the functions they can perform. 
Service accounts have been in beta since Grafana v8.5. 
During that time, we've improved the UI and migration path from API keys, made it possible to add service accounts to teams, and inherit team permissions. 
To learn more about Service Accounts, see the [documentation]({{< relref "../administration/service-accounts" >}}).
```

{{< figure src="/static/img/docs/service-accounts/add-service-account-token-9-1.png" max-width="750px" caption="Adding a service account token" >}}

#### JWT URL embedding

> You can now easily embed Grafana in other applications by adding a JWT token directly in the Grafana's URL, for example, it can look like `https://example.grafana.net/dashboard/uuid?aut_token=<jwt_token>`. When the JWT token is passed through the request URL to Grafana, Grafana will validate and authenticate the token linked to a specific user, allowing access to dashboards which that user can view. To see JWT URL embedding in action, see the [sample project](https://github.com/grafana/grafana-iframe-oauth-sample).

Reviewer: ```suggestion
You can now easily embed Grafana in other applications by adding a JWT token directly in the Grafana's URL, for example,`https://example.grafana.net/dashboard/uuid?aut_token=<jwt_token>`. 
When the JWT token is passed through the request URL to Grafana, Grafana validates and authenticates the token linked to a specific user, allowing access to dashboards which that user can view. 
To see JWT URL embedding in action, see the [sample project](https://github.com/grafana/grafana-iframe-oauth-sample).
```

{{< figure src="/static/img/docs/dashboards/jwt-url-embedding-9-1.png" max-width="750px" caption="A JWT token used to embed Grafana" >}}

#### Organization role mapping for GitHub OAuth2 authentication

You can now use GitHub OAuth2 to map users or teams to specific [Grafana organization roles]({{< relref "../administration/roles-and-permissions/#organization-roles" >}}) by using `role_attribute_path` configuration option. Grafana will use [JMESPath](https://jmespath.org/examples.html) for path lookup and role mapping. For more information, see the [documentation]({{< relref "../setup-grafana/configure-security/configure-authentication/github/#map-roles" }}).

{{< figure src="/static/img/docs/permissions/org-role-mapping-github-9-1.png" max-width="750px" caption="Configuring GitHub OAuth2 authentication with role mapping" >}}

### Search and navigation

#### (Beta) Panel title search and search improvements

> We've improved the performance of searching by panel title. If a panel's title matches your search query, it will be displayed in the search results. This feature will be rolled out to Grafana Cloud users over the course of several weeks, or can be accessed by enabling the panelTitleSearch feature toggle.

Reviewer 2: ```suggestion
We've improved the performance of searching by panel title. If a panel's title matches your search query, it will be displayed in the search results. This feature will be rolled out to Grafana Cloud users over the course of several weeks, or can be accessed by enabling the `panelTitleSearch` feature toggle.
```

{{< figure src="/static/img/docs/dashboards/panel-title-search-9-1.png" max-width="750px" caption="Searching for a panel title" >}}

#### Starred dashboards in the navigation bar

As part of the upcoming improvements to Grafana's navigation, you can now directly access your [starred dashboards]({{< relref "../dashboards/use-dashboards/" }}) from the navigation bar.

{{< figure src="/static/img/docs/dashboards/starred-dashboards-9-1.png" max-width="750px" caption="Accessing your starred dashboards" >}}

### Panels

#### Heatmap improvements

The beta heatmap announced in version 9.0 is now used throughout Grafana. Its performance is improved, and it now supports [exemplars]({{< relref "../basics/exemplars/" }}). To learn more about the heatmap panel, see the [documentation]({{< relref "../visualizations/heatmap/" >}}).

{{< figure src="/static/img/docs/panels/heatmap-panel-9-1.png" max-width="750px" caption="A heatmap panel" >}}

> #### (Beta) Trace to Metrics

Reviewer: ```suggestion
#### (Beta) Trace to metrics
```

You can now link metrics queries to your traces. This feature can be accessed by enabling the `traceToMetrics` feature toggle.

> In your tracing datasource configuration, select a metrics datasource, add tags, and write your queries. Each query will appear as a link on each span. The configured tag values will be dynamically added to your metrics queries. You can link to any metric you'd like, and metrics for span durations, counts, and errors filtered by service or span are a great starting point. The metrics generator introduced in Tempo 1.4 pairs extremely well with Trace to Metrics. To learn more about the metrics generator, see the [blog post](https://grafana.com/blog/2022/05/02/new-in-grafana-tempo-1.4-introducing-the-metrics-generator/).

Reviewer: Changed to keep things in present tense. 
```suggestion
In your tracing datasource configuration, select a metrics datasource, add tags, and write your queries.
Each query appears as a link on each span.
The configured tag values are dynamically added to your metrics queries. 
You can link to any metric you'd like. 
Metrics for span durations, counts, and errors filtered by service or span are a great starting point.
The metrics generator introduced in Tempo 1.4 pairs extremely well with Trace to Metrics. 
To learn more about the metrics generator, see the [blog post](https://grafana.com/blog/2022/05/02/new-in-grafana-tempo-1.4-introducing-the-metrics-generator/).
```

{{< figure src="/static/img/docs/panels/trace-to-metrics-9-1.png" max-width="750px" caption="Linking a trace to a metrics query" >}}

> #### (Beta) APM Table

Reviewer: ```suggestion
#### (Beta) APM table
```

You can now get APM data with Grafana. The data is shown in a handy table in the Tempo datasource under the Service Graph tab. This feature can be accessed by enabling the `tempoApmTable` feature toggle. You will need to enable the metrics generator to receive the data for the APM table.

The APM table displays RED metrics. See your top five request, error, and duration span metrics in the table summary view. The span metrics are created using the Tempo [metrics generator](https://grafana.com/blog/2022/05/02/new-in-grafana-tempo-1.4-introducing-the-metrics-generator/).

We also included several handy links, embedded directly into the table. These links direct you to a Prometheus query to further investigate the data. We also provide a link directly to Tempo search from the table, making it easier for you to investigate your APM metrics.

To learn more about the APM table, see the [documentation]({{< relref "../datasources/tempo" >}}).

{{< figure src="/static/img/docs/panels/apm-table-9-1.png" max-width="750px" caption="An APM table in the Explore view" >}}

### Sharing

> #### (Alpha) Public Dashboards

Reviewer: ```suggestion
#### (Alpha) Public dashboards
```

> #### (Alpha) Public Dashboards

Reviewer: Is the term Publish Dashboards considered a proper noun?  The capitalization seems to be inconsistent in the text below.

Author: Probably not.

Public Dashboards are available as an alpha feature that can be enabled with the `publicDashboards` feature toggle.

You can generate a link for dashboards that you'd like to share publicly. Anyone with the link will be able to access that dashboard, and nothing else.

The public view of a dashboard has a few restrictions:

> - Arbitrary queries cannot be run against your datasources through public dashboards. Public dashboards can only execute the queries stored on the original dashboard.

Reviewer: Is data sources one or two words?

Author: Two.

- The public dashboard is displayed in a read-only kiosk view.
- The time range is fixed to the dashboard default time range.

To learn more about Public Dashboards, see the [documentation]({{< relref "../dashboards/dashboard-public/" >}}).

#### Provisioning improvements for Grafana Alerting

Provision your Grafana Alerting resources directly from disk, which is a process you may already follow for your dashboards or data sources. Grafana Alerting resources are provisioned once when you set up Grafana.

Provisioning for Grafana Alerting currently supports the following resources:

- Alert rules
- Contact points
- Notification policies
- Mute timings
- Text templates

For more information, see the
[provisioning documentation]({{< relref "../administration/provisioning/" >}}).

## Grafana Enterprise

### Sharing and customization

#### Reporting improvements

Reporting is better in a few specific ways in Grafana 9.1:

- You can save a draft of a report, in case you need to make a quick update to a dashboard or make other changes before publishing the report.
- Each page of a report PDF includes the dashboard's name, which is useful for multi-page reports.

> - You can send the same dashboard in a report multiple times with different time ranges - for example sharing last month's numbers as compared to the numbers for this month. Note that the dashboard will use the same template variables if you attach it to a report twice.

Reviewer: Replacing dash with an emdash. 
```suggestion
- You can send the same dashboard in a report multiple times with different time ranges&mdash;for example sharing last month's numbers as compared to the numbers for this month. Note that the dashboard will use the same template variables if you attach it to a report twice.
```

{{< figure src="/static/img/docs/enterprise/rbac-app-plugins-9-1.png" max-width="750px" caption="Configuring role-based-access to app plugins" >}}

#### Rotate your database encryption keys using Grafana's API

In Grafana version 9.0, we [revamped]({{< relref "./whats-new-in-v9-0/#envelope-encryption-is-generally-available-and-enabled-by-default"  >}}) the method Grafana uses to encrypt secrets, like data source credentials, so that you can rotate encryption keys and integrate with a key management system like Hashicorp Vault, AWS Key Management Service, or Azure Key Vault.

Now, you can rotate keys and re-encrypt secrets via API. This makes it easier to configure Grafana to be secure while deploying it. To learn more, see our guide to [configuring database encryption]({{< relref "../setup-grafana/configure-security/configure-database-encryption/#configure-database-encryption" >}}).

#### Audit all actions in Grafana using Verbose Mode

> Auditing logs helps you manage and mitigate activity, and meet compliance requirements. By default, Grafana emits an audit log with every action that changes something (like a user creating or updating a dashboard, or updating another user's permissions). If you want to record all actions on the Grafana server, including GETs and page views, you can now turn on `verbose mode`. This results in more logs, but it can be useful to debug specific issues or make sure you catch everything happening in Grafana for security or compliance reasons. To learn more, see the [configuration documentation]({{< relref "../setup-grafana/configure-grafana/enterprise-configuration/#verbose" >}}).

Reviewer: ```suggestion
Auditing logs helps you manage and mitigate activity and meet compliance requirements. 
By default, Grafana emits an audit log with every action that changes something (like a user creating or updating a dashboard or updating another user's permissions). 
If you want to record all actions on the Grafana server, including GETs and page views, you can now turn on `verbose mode`. 
This results in more logs, but it can be useful to debug specific issues or make sure you catch everything happening in Grafana for security or compliance reasons. 
To learn more, see the [configuration documentation]({{< relref "../setup-grafana/configure-grafana/enterprise-configuration/#verbose" >}}).
```

{{< figure src="/static/img/docs/enterprise/oauth-synced-user-9-1.png" max-width="750px" caption="Non-interactive view of a user synced via OAuth" >}}

#### Wildcard support for LDAP groups in Team Sync

Team sync lets you set up synchronization between your authentication provider's teams and Grafana teams. To leverage your existing Active Directory in an efficient way without having to create multiple teams, you can now use wildcards when configuring LDAP groups, so that multiple Active Directory groups can be added at once to a single team.

{{< figure src="/static/img/docs/enterprise/ldap-wildcard-teamsync-9-1.png" max-width="750px" caption="Using wildcard mapping for team sync" >}}

> #### Redirect binding support for AzureAD SAML Single Logout

Reviewer: Is "Single Logout" a proper noun?

Author: Yes.

To learn more about authentication, see the [documentation]({{< relref "../setup-grafana/configure-security/configure-authentication/" >}}).

{{< figure src="/static/img/docs/enterprise/oauth-synced-user-9-1.png" max-width="750px" caption="Non-interactive view of a user synced via OAuth" >}}

#### Support wildcards for LDAP groups in team sync

Team sync lets you set up synchronization between your authentication provider's teams and Grafana teams.
To leverage your existing Active Directory in an efficient way without having to create multiple teams, you can now use wildcards when configuring LDAP groups, so that multiple Active Directory groups can be added at once to a single team.

{{< figure src="/static/img/docs/enterprise/ldap-wildcard-teamsync-9-1.png" max-width="750px" caption="Using wildcard mapping for team sync" >}}

#### Redirect binding support for AzureAD SAML Single Logout

SAML Single Logout allows users to log out from all applications associated with the current IdP (Identity Provider) session established via SAML Single Sign-On (SSO).
To enable integration of more use cases, we have added support for an HTTP-Redirect binding for a Single Logout.

> #### AWS Secrets Manager integration

Reviewer 3: Hi, can we drop this line item from the "what's new" section?

Reviewer 3: Dropped in https://github.com/grafana/grafana/pull/53066/commits/66e4876618f2d6faaeea8f43dfba74da88a8cd47
